manticore/core/cpu/abstractcpu.py

```diff
@@ -755,41 +803,58 @@ def emulate(self, instruction):
         '''
 
         emu = UnicornEmulator(self)
-        emu.emulate(instruction)
+        emu.emulate(insn)
 
 
         # We have been seeing occasional Unicorn issues with it not clearing
         # the backing unicorn instance. Saw fewer issues with the following
         # line present.
         del emu
 
-    def render_instruction(self):
+    def render_instruction(self, insn=None):
         try:
-            instruction = self.instruction
-            return "INSTRUCTION: 0x%016x:\t%s\t%s"%( instruction.address, instruction.mnemonic, instruction.op_str)
-        except:
-            return "{can't decode instruction }"
+            if (insn is None or
+                    not isinstance(insn, BinjaILDisasm.BinjaILInstruction)):
+                insn = self.instruction
+                return "INSTRUCTION: 0x%016x:\t%s\t%s" % (insn.address,
+                                                          insn.mnemonic,
+                                                          insn.op_str)
+            else:
+                return str(insn)
+        except Exception as e:
+            return "{can't decode instruction}"
 
     def render_register(self, reg_name):
         result = ""
 
         value = self.read_register(reg_name)
-
+        reg_name = reg_name.upper()
+        if (value == 0 or
```

Comment:
Yes, my take was that we are not really missing information on this, since not printing a register denotes that it is O. Also we don't differentiate uninitialized registers from registers with 0 values (aka all registers start with 0 instead of, say, a magic number denoting they have never been used).  Still though, not having strong opinions on this either. What about the sorting of the registers in the output?